web/src/components/PreflightResultPage.tsx

```diff
@@ -219,7 +219,7 @@ function PreflightResultPage(props: Props) {
           )}
         </div>
       )}
-      {!props.fromLicenseFlow && preflightCheck?.showPreflightCheckPending && (
+      {!props.fromLicenseFlow && preflightCheck?.shouldShowRerunPreflight && (
```

Comment:
Previously this was a giant ternary. Decided to make an explicit property from the response for this. it's a `shouldShow` because we have to combine the BE state from the response with this `fromLicenseFlow` prop. I think we can get `fromLicenseFlow` from the BE but I'm not sure what endpoint is coming from yet.

so basically the logic was 
`!this.props.fromLicenseFlow && stopPolling` 

- the `stopPolling` part been moved into the selector as `shouldShowRerunPreflight` 
- we're still checking `!this.props.fromLicenseFlow` here 

<img width="741" alt="Screenshot 2023-02-15 at 16 10 00" src="https://user-images.githubusercontent.com/4998130/219196190-b1aac721-d16f-4966-bc38-31915732342c.png">